sql/index/pilosa/driver.go

```diff
@@ -113,11 +114,13 @@ func (d *Driver) LoadAll(db, table string) ([]sql.Index, error) {
 	return indexes, err
 }
 
+var errInvalidIndexType = errors.NewKind("expecting a pilosa index, instead got %T")
+
 // Save the given index (mapping and bitmap)
-func (d *Driver) Save(ctx context.Context, idx sql.Index, iter sql.IndexKeyValueIter) error {
-	path, err := mkdir(d.root, idx.Database(), idx.Table(), idx.ID())
-	if err != nil {
-		return err
+func (d *Driver) Save(ctx context.Context, i sql.Index, iter sql.IndexKeyValueIter) error {
+	idx, ok := i.(*pilosaIndex)
```

Comment:
So far, it's OK. But I wonder what we can do with this in future. Because right now indexes and drivers will be tightly coupled (I mean you cannot use driver with index created by other driver). Maybe this is fine if we won't use multiple drivers at the same time.

Reply:
Even if we use multiple drivers, a driver should only be able to know how to save/create/etc its own drivers. You can't save a, say btree index, using the pilosa driver.